django/views/decorators/debug.py

```diff
@@ -41,6 +46,34 @@ def sensitive_variables_wrapper(*func_args, **func_kwargs):
                 sensitive_variables_wrapper.sensitive_variables = "__ALL__"
             return func(*func_args, **func_kwargs)
 
+        if iscoroutinefunction(func):
+            markcoroutinefunction(sensitive_variables_wrapper)
+
+            wrapped_func = func
+            while getattr(wrapped_func, "__wrapped__", None) is not None:
+                wrapped_func = wrapped_func.__wrapped__
+
+            try:
+                file_path = inspect.getfile(wrapped_func)
+            except TypeError:  # raised for builtins
+                file_path = None
+
+            try:
+                _file_lines, first_file_line = inspect.getsourcelines(wrapped_func)
+            except TypeError:  # raised for builtins
+                first_file_line = None
```

Comment:
I couldn't find any builtin that is a coroutine.  This check is currently only needed theoretically, am I right?

Reply:
I meant "builtins" as in any function that is implemented in C, rather than just from the `builtins` module. I'm not knowledgeable enough to say whether there are any async functions implemented in C in the standard library, but this restriction would also apply to any C/Rust/whatever extensions to Python which makes me think we should handle this case in SOME way.

I think your changes make sense, will comment on the new version with some wording tweaks